vtex-apps/product-list
# fixed: Duplicated keys when product is splitted in minicart

#### What problem is this solving?

When a product has gifts, checkout always split the item into the cart

![image](https://user-images.githubusercontent.com/13649073/219728832-1d7a94b4-5888-4e90-bd87-65e2bf8f0101.png)

A unique ID is the same for the product, it's unique for the product and not an instance of a product

![image](https://user-images.githubusercontent.com/13649073/219729631-76965d32-a271-41cc-bb98-df8dd83d8fdd.png)

Duplicated keys are a problem in React when talking about lists: https://reactjs.org/docs/lists-and-keys.html

The items element in the product list was working properly with two products (which means that the graphql and orderform response was right), the problem was that the virtual DOM was not being updated due to the duplicated keys for that product

#### How to test it?

- Add [this product](https://iespinoza--itzestore.myvtex.com/gargantillaarcoiris/p) to the cart
- It will add the product and also a gift
- Increase the quantity of this product in minicart, it will add another gift (two splitted gifts)
- Remove one quantity of the product in minicart, not the gift (the expected result is to remove one of the splitted gifts, the behavior that we are having now we keep the two gifts in minicart)

#### Screenshots or example usage:

[Before](https://master--itzestore.myvtex.com/gargantillaarcoiris/p)

<img width="383" alt="image" src="https://user-images.githubusercontent.com/13649073/219741557-12cc256a-05c6-4de1-9b72-bcdb80fbb0fe.png">


[After](https://iespinoza--itzestore.myvtex.com/gargantillaarcoiris/p)

<img width="394" alt="image" src="https://user-images.githubusercontent.com/13649073/219741083-5efda4b8-4b8c-492a-b4d2-d4acf041a6bb.png">

#### Related to / Depends on

Zendesk #715876

#### How does this PR make you feel? [:link:](http://giphy.com/)

![](https://media.giphy.com/media/xT0GqssRweIhlz209i/giphy.gif)
